### ISSUE TITLE
1.99.9-beta cannot parse requires += in a META file

### ISSUE THREAD

**Issue by arjunguha** [OCamlPro/ocp-build#2]
I have ocp-build and `ppx_deriving` installed from opam. ppx_deriving uses `requires +=` in its META file:

https://github.com/whitequark/ppx_deriving/blob/master/pkg/META.in#L27

ocp-build doesn't like that META file:

```
arjun@geppetto:tmp$ ocp-build list
find_project_root
Warning: Could not find OCaml ocamldoc tool.
iter: Don't know what to do with:
  IDENT "requires"
  PLUSEQUAL
  STRING "ppx_deriving.enum ppx_deriving.iter ppx_deriving.map"
  IDENT "requires"
  PLUSEQUAL
  STRING "ppx_deriving.fold ppx_deriving.create"
  RPAREN
  IDENT "package"
  STRING "show"
  LPAREN
  IDENT "version"
  EQUAL
  STRING "2.2"
  IDENT "description"
  EQUAL
  STRING "[@@deriving show]"
  IDENT "requires"
  LPAREN
  MINUS
  IDENT "ppx_driver"
  RPAREN
  EQUAL
  STRING "ppx_deriving"
  IDENT "ppxopt"
  LPAREN
  MINUS
  IDENT "ppx_driver"
  RPAREN
  EQUAL
  STRING "ppx_deriving,./ppx_deriving_show.cma"
  IDENT "requires"
  LPAREN
  IDENT "ppx_driver"
  RPAREN
  EQUAL
  STRING "ppx_deriving.api"
  IDENT "archive"
  LPAREN
  IDENT "ppx_driver"
  IDENT "byte"
  RPAREN
  EQUAL
  STRING "ppx_deriving_show.cma"
  IDENT "archive"
  LPAREN
  IDENT "ppx_driver"
  IDENT "native"
  RPAREN
  EQUAL
  STRING "ppx_deriving_show.cmxa"
  IDENT "exists_if"
  EQUAL
  STRING "ppx_deriving_show.cma"
  RPAREN
  IDENT "package"
  STRING "eq"
  LPAREN
  IDENT "version"
  EQUAL
  STRING "2.2"
  IDENT "description"
  EQUAL
  STRING "[@@deriving eq]"
  IDENT "requires"
  LPAREN
  MINUS
  IDENT "ppx_driver"
  RPAREN
  EQUAL
  STRING "ppx_deriving"
  IDENT "ppxopt"
  LPAREN
  MINUS
  IDENT "ppx_driver"
  RPAREN
  EQUAL
  STRING "ppx_deriving,./ppx_deriving_eq.cma"
  IDENT "requires"
  LPAREN
  IDENT "ppx_driver"
  RPAREN
  EQUAL
  STRING "ppx_deriving.api"
  IDENT "archive"
  LPAREN
  IDENT "ppx_driver"
  IDENT "byte"
  RPAREN
  EQUAL
  STRING "ppx_deriving_eq.cma"
  IDENT "archive"
  LPAREN
  IDENT "ppx_driver"
  IDENT "native"
  RPAREN
  EQUAL
  STRING "ppx_deriving_eq.cmxa"
  IDENT "exists_if"
  EQUAL
  STRING "ppx_deriving_eq.cma"
  RPAREN
  IDENT "package"
  STRING "ord"
  LPAREN
  IDENT "version"
  EQUAL
  STRING "2.2"
  IDENT "description"
  EQUAL
  STRING "[@@deriving ord]"
  IDENT "requires"
  LPAREN
  MINUS
  IDENT "ppx_driver"
  RPAREN
  EQUAL
  STRING "ppx_deriving"
  IDENT "ppxopt"
  LPAREN
  MINUS
  IDENT "ppx_driver"
  RPAREN
  EQUAL
  STRING "ppx_deriving,./ppx_deriving_ord.cma"
  IDENT "requires"
  LPAREN
  IDENT "ppx_driver"
  RPAREN
  EQUAL
  STRING "ppx_deriving.api"
  IDENT "archive"
  LPAREN
  IDENT "ppx_driver"
  IDENT "byte"
  RPAREN
  EQUAL
  STRING "ppx_deriving_ord.cma"
  IDENT "archive"
  LPAREN
  IDENT "ppx_driver"
  IDENT "native"
  RPAREN
  EQUAL
  STRING "ppx_deriving_ord.cmxa"
  IDENT "exists_if"
  EQUAL
  STRING "ppx_deriving_ord.cma"
  RPAREN
  IDENT "package"
  STRING "enum"
  LPAREN
  IDENT "version"
  EQUAL
  STRING "2.2"
  IDENT "description"
  EQUAL
  STRING "[@@deriving enum]"
  IDENT "requires"
  LPAREN
  MINUS
  IDENT "ppx_driver"
  RPAREN
  EQUAL
  STRING "ppx_deriving"
  IDENT "ppxopt"
  LPAREN
  MINUS
  IDENT "ppx_driver"
  RPAREN
  EQUAL
  STRING "ppx_deriving,./ppx_deriving_enum.cma"
  IDENT "requires"
  LPAREN
  IDENT "ppx_driver"
  RPAREN
  EQUAL
  STRING "ppx_deriving.api"
  IDENT "archive"
  LPAREN
  IDENT "ppx_driver"
  IDENT "byte"
  RPAREN
  EQUAL
  STRING "ppx_deriving_enum.cma"
  IDENT "archive"
  LPAREN
  IDENT "ppx_driver"
  IDENT "native"
  RPAREN
  EQUAL
  STRING "ppx_deriving_enum.cmxa"
  IDENT "exists_if"
  EQUAL
  STRING "ppx_deriving_enum.cma"
  RPAREN
  IDENT "package"
  STRING "iter"
  LPAREN
  IDENT "version"
  EQUAL
  STRING "2.2"
  IDENT "description"
  EQUAL
  STRING "[@@deriving iter]"
  IDENT "requires"
  LPAREN
  MINUS
  IDENT "ppx_driver"
  RPAREN
  EQUAL
  STRING "ppx_deriving"
  IDENT "ppxopt"
  LPAREN
  MINUS
  IDENT "ppx_driver"
  RPAREN
  EQUAL
  STRING "ppx_deriving,./ppx_deriving_iter.cma"
  IDENT "requires"
  LPAREN
  IDENT "ppx_driver"
  RPAREN
  EQUAL
  STRING "ppx_deriving.api"
  IDENT "archive"
  LPAREN
  IDENT "ppx_driver"
  IDENT "byte"
  RPAREN
  EQUAL
  STRING "ppx_deriving_iter.cma"
  IDENT "archive"
  LPAREN
  IDENT "ppx_driver"
  IDENT "native"
  RPAREN
  EQUAL
  STRING "ppx_deriving_iter.cmxa"
  IDENT "exists_if"
  EQUAL
  STRING "ppx_deriving_iter.cma"
  RPAREN
  IDENT "package"
  STRING "map"
  LPAREN
  IDENT "version"
  EQUAL
  STRING "2.2"
  IDENT "description"
  EQUAL
  STRING "[@@deriving map]"
  IDENT "requires"
  LPAREN
  MINUS
  IDENT "ppx_driver"
  RPAREN
  EQUAL
  STRING "ppx_deriving"
  IDENT "ppxopt"
  LPAREN
  MINUS
  IDENT "ppx_driver"
  RPAREN
  EQUAL
  STRING "ppx_deriving,./ppx_deriving_map.cma"
  IDENT "requires"
  LPAREN
  IDENT "ppx_driver"
  RPAREN
  EQUAL
  STRING "ppx_deriving.api"
  IDENT "archive"
  LPAREN
  IDENT "ppx_driver"
  IDENT "byte"
  RPAREN
  EQUAL
  STRING "ppx_deriving_map.cma"
  IDENT "archive"
  LPAREN
  IDENT "ppx_driver"
  IDENT "native"
  RPAREN
  EQUAL
  STRING "ppx_deriving_map.cmxa"
  IDENT "exists_if"
  EQUAL
  STRING "ppx_deriving_map.cma"
  RPAREN
  IDENT "package"
  STRING "fold"
  LPAREN
  IDENT "version"
  EQUAL
  STRING "2.2"
  IDENT "description"
  EQUAL
  STRING "[@@deriving fold]"
  IDENT "requires"
  LPAREN
  MINUS
  IDENT "ppx_driver"
  RPAREN
  EQUAL
  STRING "ppx_deriving"
  IDENT "ppxopt"
  LPAREN
  MINUS
  IDENT "ppx_driver"
  RPAREN
  EQUAL
  STRING "ppx_deriving,./ppx_deriving_fold.cma"
  IDENT "requires"
  LPAREN
  IDENT "ppx_driver"
  RPAREN
  EQUAL
  STRING "ppx_deriving.api"
  IDENT "archive"
  LPAREN
  IDENT "ppx_driver"
  IDENT "byte"
  RPAREN
  EQUAL
  STRING "ppx_deriving_fold.cma"
  IDENT "archive"
  LPAREN
  IDENT "ppx_driver"
  IDENT "native"
  RPAREN
  EQUAL
  STRING "ppx_deriving_fold.cmxa"
  IDENT "exists_if"
  EQUAL
  STRING "ppx_deriving_fold.cma"
  RPAREN
  IDENT "package"
  STRING "create"
  LPAREN
  IDENT "version"
  EQUAL
  STRING "2.2"
  IDENT "description"
  EQUAL
  STRING "[@@deriving create]"
  IDENT "requires"
  LPAREN
  MINUS
  IDENT "ppx_driver"
  RPAREN
  EQUAL
  STRING "ppx_deriving"
  IDENT "ppxopt"
  LPAREN
  MINUS
  IDENT "ppx_driver"
  RPAREN
  EQUAL
  STRING "ppx_deriving,./ppx_deriving_create.cma"
  IDENT "requires"
  LPAREN
  IDENT "ppx_driver"
  RPAREN
  EQUAL
  STRING "ppx_deriving.api"
  IDENT "archive"
  LPAREN
  IDENT "ppx_driver"
  IDENT "byte"
  RPAREN
  EQUAL
  STRING "ppx_deriving_create.cma"
  IDENT "archive"
  LPAREN
  IDENT "ppx_driver"
  IDENT "native"
  RPAREN
  EQUAL
  STRING "ppx_deriving_create.cmxa"
  IDENT "exists_if"
  EQUAL
  STRING "ppx_deriving_create.cma"
  RPAREN
Warning: exception "Failure(\"Unexpected tokens\")" while loading "/home/arjun/.opam/system/lib/ppx_deriving/META"
Warning: 2 missing and 99 disabled (use -print-missing)
```

**Comment by AltGr** [comment]
this makes ocp-build unusable whenever ppx_deriving is installed: upon encountering the META file it cannot parse, besides being verbose, it lists all packages as "incomplete" and doesn't attempt to build them.

All that with only warnings... and while returning 0.

**Comment by AltGr** [comment]
Oops, sorry: it seems that was a different issue that made the build fail. My bad.

**Comment by lefessan** [comment]
Fixed in 1.99.13-beta
